include/core/thermal_conductivity_model.h

```diff
@@ -188,4 +188,122 @@ class ThermalConductivityLinear : public ThermalConductivityModel
   const double B;
 };
 
+/**
+ * @brief ThermalConductivityPhaseChange Implements a phase-dependant thermal conductivity
+ */
+class ThermalConductivityPhaseChange : public ThermalConductivityModel
+{
+public:
+  /**
+   * @brief Default constructor
+   */
+  ThermalConductivityPhaseChange(const double thermal_conductivity_s,
+                                 const double thermal_conductivity_l,
+                                 const double T_solidus,
+                                 const double T_liquidus)
+    : thermal_conductivity_s(thermal_conductivity_s)
+    , thermal_conductivity_l(thermal_conductivity_l)
+    , T_solidus(T_solidus)
+    , T_liquidus(T_liquidus)
+  {}
```

Comment:
You need to add to the constructor that     this->model_depends_on[field::temperature] = true;
Look at the ThermalConductivityLinear model for an example.

Reply:
done.